cupy/cuda/texture.pyx

```diff
@@ -426,11 +428,16 @@ cdef class CUDAarray:
             param = <Memcpy3DParms*>self._make_cudaMemcpy3DParms(arr, self)
         elif direction == 'out':
             param = <Memcpy3DParms*>self._make_cudaMemcpy3DParms(self, arr)
+
+        # we need to serialize on the current or given stream to ensure
+        # data is ready
+        cdef intptr_t stream_ptr
         try:
             if stream is None:
-                runtime.memcpy3D(<intptr_t>param)
+                stream_ptr = stream_module.get_current_stream_ptr()
             else:
-                runtime.memcpy3DAsync(<intptr_t>param, stream.ptr)
+                stream_ptr = <intptr_t>(stream.ptr)
+            runtime.memcpy3DAsync(<intptr_t>param, stream_ptr)
```

Comment:
Strictly speaking this slightly changes the semantics specified in the docstrings of `copy_to`/`copy_from` below. Should I make it a breaking change by revising the docstrings (emphasize it's always asynchronous)?

Reply:
Alternatively, we can leave `texture.pyx` as is by explicitly synchronizing the current stream in the tests, but I think users wouldn't anticipate this is needed, so let me know which solution is preferred:
1. This PR + emphasizing asynchronous copies
2. Fix `test_texture.py` only

Reply:
I think this PR is good